File: code-samples/ai/code-snippets/speaker-diarization.js
const fs = require ('fs')
var platform = require('./../quick-start.js').platform;
NGROK_ADDRESS = process.env.NGROK_URL // "http://d973-73-170-11-87.ngrok-free.app"
WEBHOOK_URL = NGROK_ADDRESS + "/webhook";
CONTENT_URI = 'https://rclabs-addin-resources.s3.us-east-1.amazonaws.com/media/Sample%20Call%20Tech%20Support.mp3'
speakers_recogition()


// Next line must be at the 10th line!
platform.on(platform.events.loginSuccess, () => {
    NGROK_ADDRESS = "NGROK-TUNNEL-ADDRESS"
    WEBHOOK_URL = NGROK_ADDRESS + "/webhook";
    CONTENT_URI = "PUBLICLY-ACCESSIBLE-CONTENT-URI"
    speakers_recogition()
})

platform.on(platform.events.loginError, function(e){
    console.log("Unable to authenticate to platform. Check credentials.", e.message)
    process.exit(1)
});

/*
* Recognize speakers from a conversation
*/
async function speakers_recogition() {
    try {
        let bodyParams = {
            contentUri:   CONTENT_URI,
            encoding:     "Mpeg",
            languageCode: "en-US",
            source:       "RingCentral",
            audioType:    "CallCenter"
        }
        let endpoint = `/ai/audio/v1/async/speaker-diarize?webhook=${WEBHOOK_URL}`
        let resp = await platform.post(endpoint, bodyParams);
        let jsonObj = await resp.json();
        if (resp.status == 202) {
          console.log("Job ID: " + jsonObj.jobId);
          console.log("Ready to receive response at: " + WEBHOOK_URL);
        }
    } catch (e) {
          console.log(`Unable to call speaker diarization API. ${e.message}`);
    }
}

File: code-samples/ai/code-snippets/speaker-diarization.java
package SpeakersDiarization;

import java.io.IOException;
import com.google.common.reflect.TypeToken;
import com.google.gson.Gson;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class SpeakersDiarization {
    String NGROK_ADDRESS = "NGROK-TUNNEL-ADDRESS";
    String WEBHOOK_URL = NGROK_ADDRESS + "/webhook";
    String CONTENT_URI = "PUBLICLY-ACCESSIBLE-CONTENT-URI";

    static RestClient restClient;

    public static void main(String[] args) {
      var obj = new SpeakersDiarization();
      try {
        // Instantiate the SDK
        restClient = new RestClient("PRODUCTION-APP-CLIENT-ID", "PRODUCTION-APP-CLIENT-SECRET", "https://platform.ringcentral.com");

        // Authenticate a user using a personal JWT token
        restClient.authorize("PRODUCTION-JWT");

        obj.speakers_detection();

      } catch (RestException e) {
        System.out.println(e.getMessage());
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
    /*
    * Detect speakers from a conversation
    */
    static private void speakers_detection() throws RestException, IOException {
      try {
        var bodyParams = new DiarizeInput()
        .contentUri(CONTENT_URI)
        .encoding("Mpeg")
        .languageCode("en-US")
        .source("RingCentral")
        .audioType("CallCenter");

        var callbackAddress = System.getenv("WEBHOOK_URL") + "/webhook";
        var queryParams = new CaiSpeakerDiarizeParameters().webhook(callbackAddress);
        var resp = restClient.ai().audio().v1().async().speakerDiarize().post(bodyParams, queryParams);
        System.out.println("Job ID: " + resp.jobId);
        System.out.println("Ready to receive response at: " + callbackAddress);
      }catch (RestException ex) {
        System.out.println("Unable to call speaker diarization API. " + ex.getMessage());
      }
    }
}
